### PR TITLE
Add setup information, add async-mutex to package.json

### PR DESCRIPTION
**Pull request by wesen**

### CODE REVIEW

**Comment by wesen** [comment]
I was missing the async-mutex package. 

The instructions are not entirely complete yet, I haven't entirely figured out how to setup the practice room channels.

File: README.md [hunk 1]
```diff
@@ -16,3 +16,30 @@ if you just want to jump in without install `nodenv`, make sure you look at [`.n
 for the current version we are using.
 
 PS - You must have mongodb installed and running.
+
+## Development setup
+
+To test your own instance of the bot, do the following:
+
+* Create your own personal testing server
```
Comment (pianiststickman) [inline README.md]: Probably better just to add it to the test discord.

File: README.md [hunk 2]
```diff
@@ -16,3 +16,30 @@ if you just want to jump in without install `nodenv`, make sure you look at [`.n
...
+* [Enable developer mode](https://support.discordapp.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-)
+ in discord to be able to get guild IDs and client IDs
+* [Create an application](https://discordjs.guide/preparations/setting-up-a-bot-application.html)
+ for your bot
+* [Add the bot to your server](https://discordjs.guide/preparations/adding-your-bot-to-servers.html#bot-invite-links)
+* [Create a role for your bot](https://support.discordapp.com/hc/en-us/articles/206029707-How-do-I-set-up-Permissions-)
```
Comment (pianiststickman) [inline README.md]: IIRC this happens automatically - bots automatically get assigned a role by the integration.

**Comment by pianiststickman** [comment]
Actually, I suspect we'll soon be able to remove async-mutex entirely.

The reason we have async-mutex is because of bug #46, a possible race condition in the VC-autocreation code. The way we add rooms is to read the doc from the DB, push the new room into the array, then write the whole doc back into the DB. If the `voiceStateChanged` handler triggers twice in quick succession, one handler may read the list of rooms between when the other handler reads the list and when it writes it back to the DB with the new room - thus squashing that new room when the second handler writes *its* list back to the DB. My solution to this was to slap a mutex on the entire event handler, which met with limited success due to asynchronicity (though we haven't seen it for a while).

As it turns out the correct solution is probably to stop using `db.collections.update()` (which, per Mongo's warnings, are deprecated) and instead use `db.collections.updateOne()`, which contains an `$addToSet` atomic operator that seems like it was designed for precisely our purposes.

**Comment by sit** [comment]
Proposed changes have been superseded by other fixes.

**Pull request closed without merging**